src/front/zokrates/term.rs

```diff
@@ -19,24 +19,37 @@ lazy_static! {
     .unwrap();
     /// The modulus for ZoKrates, as an ARC
     pub static ref ZOKRATES_MODULUS_ARC: Arc<Integer> = Arc::new(ZOKRATES_MODULUS.clone());
+    /// The modulus for ZoKrates, as an IR sort
+    pub static ref ZOK_FIELD_SORT: Sort = Sort::Field(ZOKRATES_MODULUS_ARC.clone());
 }
 
 #[derive(Clone, PartialEq, Eq)]
 pub enum Ty {
     Uint(usize),
     Bool,
     Field,
-    Struct(String, BTreeMap<String, Ty>),
+    Struct(String, Vec<(String, Ty)>),
     Array(usize, Box<Ty>),
 }
 
+fn lookup<'a>(map: &'a Vec<(String, Ty)>, key: &str) -> Option<(usize, &'a Ty)> {
+    let idx = map.binary_search_by_key(&key, |p| p.0.as_str()).ok()?;
```

Comment:
So I guess this means there's an invariant that map is sorted by String, right? Should this be documented in the definition of `Ty`? Are there `debug_assert`s that would help with this? (Is this paranoia?)

Reply:
Ah, I see, `new_struct` ensures this. Would it perhaps make sense to make the `Vec<(String, Ty)>` a private wrapper type that can *only* be constructed with a function that enforces the invariant? Because as it stands now, I can construct a Struct that is broken, no?

Reply:
Yeah, I was being a little lazy here. I've split it out into its own structure and module.